tests/chainer_tests/links_tests/connection_tests/test_convolution_nd.py

```diff
@@ -26,37 +24,125 @@
     'in_channels': [4, None, 'omit'],
     'groups': [1, 2],
 })))
-class TestConvolutionND(unittest.TestCase):
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [{}]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + [
+        {'use_chainerx': True, 'chainerx_device': 'native:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+    ])
+class TestConvolutionND(testing.LinkTestCase):
+
+    param_names = ('W', 'b')
 
     def setUp(self):
-        ndim = len(self.dims)
-        self.ksize = (3,) * ndim
-        self.stride = (2,) * ndim
-        self.pad = (1,) * ndim
+        self.ndim = len(self.dims)
+        self.ksize = (3,) * self.ndim
+        self.stride = (2,) * self.ndim
+        self.pad = (1,) * self.ndim
+
+        self.x_shape = (2, 4) + self.dims
+
+        self.check_backward_options.update({'eps': 1e-2,
+                                            'atol': 1e-3, 'rtol': 1e-3})
+        if self.dtype == numpy.float16:
+            self.check_forward_options.update({'atol': 5e-4, 'rtol': 5e-3})
```

Comment:
How about 5e-3 and 5e-2 respectively to match other functions?

Reply:
Yes, I copied from the respective convolution_nd function.
I adjust it now